src/Octoshift/SecretScanningAlertService.cs

```diff
@@ -0,0 +1,163 @@
+
+            var sourceAlerts = await GetAlertsWithLocations(_sourceGithubApi, sourceOrg, sourceRepo);
+            var targetAlerts = await GetAlertsWithLocations(_targetGithubApi, targetOrg, targetRepo);
+            
+            _log.LogInformation($"Source {sourceOrg}/{sourceRepo} secret alerts found: {sourceAlerts.Count}");
+            _log.LogInformation($"Target {targetOrg}/{targetRepo} secret alerts found: {targetAlerts.Count}");
+            
+            _log.LogInformation("Matching secret resolutions from source to target repository");
+            foreach (var alert in sourceAlerts)
+            {
+                _log.LogInformation($"Processing source secret {alert.Alert.Number}");
+                
+                if (alert.Alert.State == "resolved")
+                {
+                    _log.LogInformation("  secret is resolved, looking for matching detection in target...");
+                    var target = MatchTargetSecret(alert, targetAlerts);
+
+                    if (target == null)
+                    {
+                        _log.LogWarning($"Failed to locate a matching secret to source secret {alert.Alert.Number} in {targetOrg}/{targetRepo}");
+                    }
+                    else
+                    {
+                        _log.LogInformation($"Source secret alert matched alert to {target.Alert.Number} in {targetOrg}/{targetRepo}.");
+
+                        if (alert.Alert.Resolution == target.Alert.Resolution
+                            && alert.Alert.State == target.Alert.State)
+                        {
+                            _log.LogSuccess("Source and Target Alerts are already aligned.");                            
+                        }
+                        else
+                        {
+                            _log.LogInformation($"Updating target alert:{target.Alert.Number} to state:{alert.Alert.State} and resolution:{alert.Alert.Resolution}");
+
+                            if (dryRun)
+                            {
+                                _log.LogInformation($"executing in dry run mode! Secret Alert, {target.Alert.Number}, in repository {targetOrg}/{targetRepo} would have been updated to resolution, {alert.Alert.Resolution}");
+                            }
+                            else
+                            {
+                                await _targetGithubApi.UpdateSecretScanningAlert(targetOrg, targetRepo, target.Alert.Number,
+                                    alert.Alert.State, alert.Alert.Resolution);
+                                _log.LogSuccess($"Source and Target Alert state and resolution have been aligned to {alert.Alert.Resolution}.");
+                            }   
+                        }
+                    }
+                }
+                else
+                {
+                    _log.LogInformation("  secret alert is still open, nothing to do");    
+                }
```

Comment:
IMO it's better to invert the if-else here in favor of less indentation:
```c#
if (state is not "resolved")
{
    _log.LogInformation(" secret alert is still open, nothing to do");
    return;
}

... 
```

Reply:
This is inside a for loop, return would stop the processing of the subsequent secret alerts.

Reply:
Oh yeah sorry my bad. So then the `return` should be `continue`.